src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -138,22 +140,34 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
     };
     this.cytoscapeReactWrapperRef = React.createRef();
     this.contextMenuRef = React.createRef<CytoscapeContextMenuWrapper>();
+
+    this.state = {
+      isLoading: this.props.dataSource.isLoading,
+      isError: this.props.dataSource.isError
+    };
   }
 
   componentDidMount() {
     this.cyInitialization(this.getCy());
+
+    // Listen to data source events.
+    this.props.dataSource.on('loadStart', this.loadStartHandler);
```

Comment:
My only comment is that it is a major architecture change away from Redux dispatching to this event emitter stuff. So we probably need to discuss it as a full team here.

Reply:
The EventEmitter style is more of a Node.js style than browser client style architecturally.

Reply:
Yes, indeed this is the reason I'm asking for a quick preliminary review. It's subject to change per your feedback.
If @jshaughn can do his quick review, we could have a discussion today. Then, discuss tomorrow in the water cooler with the rest of the team.